src/main/java/de/tum/in/www1/artemis/web/rest/AttachmentResource.java

```diff
@@ -122,8 +135,30 @@ public AttachmentResource(AttachmentRepository attachmentRepository, AttachmentS
     @DeleteMapping("/attachments/{id}")
     @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
     public ResponseEntity<Void> deleteAttachment(@PathVariable Long id) {
-        log.debug("REST request to delete Attachment : {}", id);
-        attachmentRepository.deleteById(id);
-        return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(ENTITY_NAME, id.toString())).build();
+        User user = userService.getUserWithGroupsAndAuthorities();
+        Optional<Attachment> optionalAttachment = attachmentRepository.findById(id);
+        if (!optionalAttachment.isPresent()) {
+            return ResponseEntity.notFound().build();
+        }
+        Attachment attachment = optionalAttachment.get();
+        Course course = null;
+        if (attachment.getLecture() != null) {
+            course = attachment.getLecture().getCourse();
+        }
+        else if (attachment.getExercise() != null) {
+            course = attachment.getExercise().getCourse();
+        }
+        if (course == null) {
+            return ResponseEntity.badRequest().build();
+        }
+        Boolean hasCourseInstructorAccess = courseService.userHasAtLeastInstructorPermissions(course);
+        if (hasCourseInstructorAccess) {
+            log.debug("REST request to delete Attachment : {}", id);
```

Comment:
please use log.info here and write the username into the log. Also add the attachment name and to which lecture (title) or exercise (title) it belongs so that we can recognize this more easily